matchms/exporting/save_as_mzspeclib.py

```diff
@@ -0,0 +1,201 @@
+    'precursor_mz': 'MS:1003208|experimental precursor monoisotopic m/z',
+    'scans': 'MS:1003057|scan number',
+    'charge': 'MS:1000041|charge state',
+    'ionmode': 'MS:1000465|scan polarity'
+}
+
+MAPPED_SPECTRUM_ATTRIBUTES = {
+    'MS:1000465|scan polarity': {
+        'positive': 'MS:1000130|positive scan',
+        'negative': 'MS:1000129|negative scan'
+    }
+}
+
+STANDARDIZED_SPECTRUM_ATTRIBUTES = {
+    'collision_energy': ['MS:1000045|collision energy', 'UO:0000266|electronvolt'],
+}
+
+
+def save_as_mzspeclib(spectra: List[Spectrum], filename: str) -> None:
+    """
+    Save a list of spectra to a file in mzSpecLib format.
+
+    Parameters:
+    spectra (List[Spectrum]): List of Spectrum objects to save.
+    filename (str): The name of the file to save the spectra to.
+    """
+    with open(filename, 'w', encoding='UTF-8') as file:
+        _write_header(filename, file)
+        for idx, spectrum in enumerate(spectra):
+            _write_spectrum(file, idx, spectrum)
+
+def _write_spectrum(file: TextIO, idx: int, spectrum: Spectrum) -> None:
+    """
+    Write a single spectrum to the file.
+
+    Parameters:
+    file (TextIO): The file object to write to.
+    idx (int): The index of the spectrum in the list.
+    spectrum (Spectrum): The Spectrum object to write.
+    """
+    print(f'<Spectrum={idx + 1}>', file=file)
+    _write_spectrum_attributes(file, spectrum)
+    if _has_analyte(spectrum):
+        _write_analyte(file, spectrum)
+    _write_peaks(file, spectrum)
+    print('', file=file)
+
+def _write_analyte(file: TextIO, spectrum: Spectrum) -> None:
+    """
+    Write analyte information for a spectrum to the file.
+
+    Parameters:
+    file (TextIO): The file object to write to.
+    spectrum (Spectrum): The Spectrum object containing analyte information.
+    """
+    print('<Analyte=1>', file=file)
+    for key, attribute in ANALYTE_ATTRIBUTES.items():
+        value = spectrum.get(key)
+        if value is not None:
+            print(f'{attribute}={value}', file=file)
+
+def _write_spectrum_attributes(file: TextIO, spectrum: Spectrum) -> None:
+    """
+    Write spectrum attributes to the file.
+
+    Parameters:
+    file (TextIO): The file object to write to.
+    spectrum (Spectrum): The Spectrum object containing attributes.
+    """
+    _write_defined_spectrum_attributes(file, spectrum)
+    
+    spectrum_attributes = spectrum.metadata.keys()
+    attr_counter = 1
+    
+    for attr in spectrum_attributes:
+        if attr in STANDARDIZED_SPECTRUM_ATTRIBUTES:
+            _write_spectrum_attribute_with_unit(file, spectrum, attr_counter, attr)
+            attr_counter += 1
+        elif attr not in SPECTRUM_ATTRIBUTES and attr not in ANALYTE_ATTRIBUTES:
+            _write_other_spectrum_attribute(file, spectrum, attr_counter, attr)
+            attr_counter += 1
+
+    print(f'MS:1003059|number of peaks={len(spectrum.peaks)}', file=file)
```

Comment:
Should this print statement still be converted to adding the number of peaks to the file?

Reply:
Never mind, I wasn't aware that it was possible to print to a file. But as long as that doesn't give performance issues, it is of course fine!